### PR TITLE
Price summery added

### PR DESCRIPTION
**Pull request by animeshgarai09**
Features added in this commit.

- Added cart price summary details.
- Shop page Ui is created.
- Created a new cart handler to remove all items.
- Scroll to the top on route change.

### CODE REVIEW

File: src/pages/shop/shop.js [hunk 1]
```diff
@@ -1,8 +1,85 @@
...
+                        </ListItem>
+                        <ListItem>
+                            <input type="radio" id="t-option" name="selector" />
+                            <label for="t-option">1 Star & above</label>
+                        </ListItem>
+                    </List>
+                </div>
+            </aside>
+            <div className={styles.product__container}>
+                {products.map((item, i) => {
```
Comment (foolhardy21) [inline src/pages/shop/shop.js:56]: suggestion - you may remove `i`its not getting used.

File: src/store/providers/basket-provider.js [hunk 2]
```diff
@@ -47,13 +47,26 @@ const BasketProvider = ({ children }) => {
...
+            toast({
+                status: "success",
+                message: "Removed all items from basket",
+                duration: 2
+            })
+        }).catch(err => {
+            console.log(err);
+        })
+    }
     // const changeCartQty = async (quantity, id) => {
```
Comment (foolhardy21) [inline src/store/providers/basket-provider.js:70]: commented code can be removed.

File: src/store/providers/product-provider.js [hunk 3]
```diff
@@ -3,17 +3,32 @@ import { ProductReducer } from '../reducers/product-reducer'
...
+                const [products, categories] = allData
+                console.log(products, categories)
                 productDispatch({
                     type: "SET_PRODUCTS",
-                    payload: response.data.products
+                    payload: products.data.products
+                })
+                productDispatch({
+                    type: "SET_CATEGORIES",
+                    payload: categories.data.categories
```
Comment (partha8) [inline src/store/providers/product-provider.js:28]: appreciation: like how you are calling for products and categories in one go.

File: src/store/providers/product-provider.js [hunk 4]
```diff
@@ -3,17 +3,32 @@ import { ProductReducer } from '../reducers/product-reducer'
...
+        "/api/products",
+        "/api/categories",
+    ];
     useEffect(() => {
-        axios.get("/api/products")
-            .then((response) => {
+        Promise.all(endpoints.map((endpoint) => axios.get(endpoint))).then(
+            axios.spread((...allData) => {
+                const [products, categories] = allData
+                console.log(products, categories)
```
Comment (foolhardy21) [inline src/store/providers/product-provider.js:21]: `console.log` may be removed if not required.

**Review by foolhardy21 — APPROVED** [review]
Appreciation - good use of the component library. nice work.

**Review by partha8 — APPROVED** [review]
Code looks good, great UI as always, and remove all function in basket page is great. I will also try to add it in future implementations of my project.

**Pull request merged**